internal/builtin/data_http.go

```diff
@@ -0,0 +1,271 @@
+package builtin
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log/slog"
+	"mime"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/blackstork-io/fabric/internal/builtin/utils"
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+)
+
+func makeHTTPDataSource(version string) *plugin.DataSource {
+	return &plugin.DataSource{
+		DataFunc: fetchHTTPDataWrapper(version),
+		Args: dataspec.ObjectSpec{
+			&dataspec.AttrSpec{
+				Name:        "url",
+				Type:        cty.String,
+				ExampleVal:  cty.StringVal("https://example.localhost/file.json"),
+				Constraints: constraint.RequiredMeaningful,
+				Doc:         "URL to fetch data from. Supported schemas are `http` and `https`",
+			},
+			&dataspec.AttrSpec{
+				Name:       "method",
+				Type:       cty.String,
+				DefaultVal: cty.StringVal("GET"),
+				OneOf: []cty.Value{
+					cty.StringVal("GET"),
+					cty.StringVal("POST"),
+					cty.StringVal("HEAD"),
+				},
+				Doc: "HTTP method for the request. Allowed methods are `GET`, `POST` and `HEAD`",
+			},
+			&dataspec.AttrSpec{
+				Name:       "basic_auth_username",
```

Comment:
Since we can do objects and blocks now as attributes, might be a bit cleaner to do:
```hcl
basic_auth {
   username = ""
   password = ""
}
# or
basic_auth = {
   username = ""
   password = ""
}
```
Where `basic_auth` attribute is optional while both `username` & `password` are required.

Reply:
indeed! I'll copy the pattern @Andrew-Morozko [implemented](https://github.com/blackstork-io/fabric/pull/136) in `data.rss` and use a `basic_auth` block